### PR TITLE
Post branching fo 4.2.x, and cleanup 

### PR DESCRIPTION
**Pull request by bsipocz**
The changelog has been pushed directly to master along with the new tag, but there is no reason not to PR the what's new, especially that it would not hurt to run the docs build to make sure all the links are functional.

The PR also does some post 4.1 release cleanup.

### CODE REVIEW

File: docs/whatsnew/4.3.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
+.. doctest-skip-all
+
+.. _whatsnew-4.3:
+
+**************************
+What's New in Astropy 4.2?
```
Comment (pllim) [inline docs/whatsnew/4.3.rst:6]:
```suggestion
What's New in Astropy 4.3?
```

File: docs/whatsnew/4.3.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Astropy 4.3 is a major release that ...  since
+the 4.2.x series of releases.
+
+In particular, this release includes:
+
+*
+*
+
+In addition to these major changes, Astropy v4.2 includes a large number of
```
Comment (pllim) [inline docs/whatsnew/4.3.rst:20]:
```suggestion
In addition to these major changes, Astropy v4.3 includes a large number of
```

File: docs/whatsnew/4.3.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+* xxx issues have been closed since v4.2
+* xxx pull requests have been merged since v4.2
+* xxx distinct people have contributed code
+
+
+Full change log
+===============
+
+To see a detailed list of all changes in version v4.2, including changes in
```
Comment (pllim) [inline docs/whatsnew/4.3.rst:32]:
```suggestion
To see a detailed list of all changes in version v4.3, including changes in
```

**Review by pllim — COMMENTED** [review]
I think there are some minor typos?

**Comment by bsipocz** [comment]
Haha, epic mess up. Thanks!

**Comment by pllim** [comment]
Is this warning expected?
```
docs/index.rst:50: WARNING: toctree contains reference to document 'whatsnew/4.1' that doesn't have a title: no link will be generated
```

**Comment by pllim** [comment]
Ah, I think this needs updating.

https://github.com/astropy/astropy/blame/master/docs/index.rst#L54

**Comment by pllim** [comment]
I see the doc is building at https://readthedocs.org/projects/astropy/builds/12200344/ but somehow the PR CI status isn't updated. 🤷

**Review by pllim — APPROVED** [review]
LGTM now. Thanks!

**Pull request merged**